framework/src/vectorpostprocessors/ElementsAlongLine.C

```diff
@@ -47,9 +47,8 @@ ElementsAlongLine::execute()
   std::vector<Elem *> intersected_elems;
   std::vector<LineSegment> segments;
 
-  MooseSharedPointer<PointLocatorBase> plb = MooseSharedPointer<PointLocatorBase>(_fe_problem.mesh().getMesh().sub_point_locator().release());
-
-  Moose::elementsIntersectedByLine(_start, _end, _fe_problem.mesh(), plb, intersected_elems, segments);
+  UniquePtr<PointLocatorBase> pl = _fe_problem.mesh().getPointLocator();
+  Moose::elementsIntersectedByLine(_start, _end, _fe_problem.mesh(), pl, intersected_elems, segments);
```

Comment:
Luckily respected coders like Herb Sutter, and others have my back:
https://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/
https://www.chromium.org/developers/smart-pointer-guidelines
http://timoch.com/blog/2013/04/std-unique_ptr-semantic/
https://books.google.com/books?id=V1kPDAAAQBAJ&pg=PA118&dq=guntheroth+unique_ptr&hl=en&sa=X&ved=0ahUKEwiw1-zBmbXOAhVW92MKHQr6D54Q6AEIHjAA#v=onepage&q=guntheroth%20unique_ptr&f=false